### PR TITLE
feat: Transaction broadcasting

### PR DESCRIPTION
**Pull request by wal33d006**

### CODE REVIEW

File: packages/transaction_signing_gateway/lib/alan/alan_transaction_broadcaster.dart [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import 'package:transaction_signing_gateway/model/transaction_broadcasting_failure.dart';
+import 'package:transaction_signing_gateway/model/transaction_hash.dart';
+import 'package:transaction_signing_gateway/transaction_broadcaster.dart';
+
+class AlanTransactionBroadcaster implements TransactionBroadcaster {
+  @override
+  Future<Either<TransactionBroadcastingFailure, TransactionHash>> broadcast(
+      {required SignedTransaction transaction, required PrivateWalletCredentials privateWalletCredentials}) async {
+    if (transaction is! SignedAlanTransaction) {
+      return left(AlanTransactionBroadcastingFailure('passed transaction is not $UnsignedAlanTransaction'));
```
Comment (andrzejchm) [inline packages/transaction_signing_gateway/lib/alan/alan_transaction_broadcaster.dart:16]:
```suggestion
      return left(AlanTransactionBroadcastingFailure('passed transaction is not $SignedAlanTransaction'));
```

File: packages/transaction_signing_gateway/lib/model/transaction_broadcasting_failure.dart [hunk 2]
```diff
@@ -0,0 +1,5 @@
+abstract class TransactionBroadcastingFailure {}
```
Comment (andrzejchm) [inline packages/transaction_signing_gateway/lib/model/transaction_broadcasting_failure.dart:1]: we'd need a `type` here, if possible. similar to what we have in `TransactionSigningFailure` in order to allow end users to `switch` over possible broadcasting errors.

**Review by andrzejchm — CHANGES_REQUESTED** [review]
two small things  :)

File: packages/transaction_signing_gateway/lib/alan/alan_transaction_broadcaster.dart (hunk 1 above)
Comment (wal33d006) [reply packages/transaction_signing_gateway/lib/alan/alan_transaction_broadcaster.dart]: ✅

File: packages/transaction_signing_gateway/lib/model/transaction_broadcasting_failure.dart (hunk 2 above)
Comment (wal33d006) [reply packages/transaction_signing_gateway/lib/model/transaction_broadcasting_failure.dart]: ✅

**Review by andrzejchm — APPROVED** [review]
🚀 let's 🚢 it!

**Pull request merged**